crates/fuel-core/src/schema/coins.rs

```diff
@@ -233,55 +298,126 @@ impl CoinQuery {
         //  https://github.com/FuelLabs/fuel-core/issues/2343
         query_per_asset.truncate(max_input as usize);
 
-        let owner: fuel_tx::Address = owner.0;
-        let query_per_asset = query_per_asset
-            .into_iter()
-            .map(|e| {
-                AssetSpendTarget::new(
-                    e.asset_id.0,
-                    e.amount.0,
-                    e.max
-                        .and_then(|max| u16::try_from(max.0).ok())
-                        .unwrap_or(max_input)
-                        .min(max_input),
-                )
-            })
-            .collect_vec();
-        let excluded_ids: Option<Vec<_>> = excluded_ids.map(|exclude| {
-            let utxos = exclude
-                .utxos
-                .into_iter()
-                .map(|utxo| coins::CoinId::Utxo(utxo.into()));
-            let messages = exclude
-                .messages
-                .into_iter()
-                .map(|message| coins::CoinId::Message(message.into()));
-            utxos.chain(messages).collect()
-        });
-
-        let base_asset_id = params.base_asset_id();
-        let spend_query =
-            SpendQuery::new(owner, &query_per_asset, excluded_ids, *base_asset_id)?;
+        let read_view = ctx.read_view()?;
+        let indexation_available = read_view.coins_to_spend_indexation_enabled;
+        if indexation_available {
```

Comment:
Maybe we can do the same as for balance where we have a separate function for old way of calculation and new way of calculation?=)

Reply:
Updated in 8319abd36c40e20093da15c046a79f8cf72c25db